vivisect/tools/graphutil.py

```diff
@@ -815,10 +789,11 @@ def getFuncCbRoutedPaths(self, fromva, tova, loopcnt=0, maxpath=None, maxsec=Non
 
         todo = [(frcbva, pnode), ]
 
-        if maxsec:
-            self.watchdog(maxsec)
-
+        starttime = time.time()
         while todo:
+            if maxsec and (time.time() - starttime > maxsec):
```

Comment:
This strikes me as "tight loop" code.  As such, we should avoid doing math inside the loop.
```
maxtime = None
if timeout:
    maxtime = time.time() + timeout    # timeout is really maxsec, but we should rename it.

while todo:
    if maxtime and time.time() > maxtime:
        raise PathForceQuitException()
```